src/main.cpp

```diff
@@ -44,13 +36,50 @@ class input_t {
 	std::set<key_t> m_held;
 };
 
-void tick(world_clock_t& out_clock, input_t const& input, sec_t dt) noexcept {
-	static constexpr f32 scale = 10.0f;
-	f32 delta{};
-	if (input.any_held(key_t::Down, key_t::Left, key_t::S, key_t::A)) { delta += -1.0f; }
-	if (input.any_held(key_t::Up, key_t::Right, key_t::W, key_t::D)) { delta += 1.0f; }
-	out_clock += scale * delta * dt.count();
-}
+class clock_ticker_t {
+  public:
+	clock_ticker_t(input_t const* input, world_clock_t* clock) noexcept : m_input(input), m_clock(clock) {}
+
+	void operator()(sec_t dt) noexcept {
+		if (m_input->any_held(key_t::Enter, key_t::R)) { update(flag::zeroing); }
+		static constexpr f32 translation_scale = 10.0f;
+		if (test(flag::zeroing)) {
+			static constexpr f32 zeroing_scale = 5.0f;
+			hour_t const delta = zeroing_scale * translation_scale * dt.count();
+			if (m_offset.hour() > delta) {
+				add(-delta);
+			} else {
+				add(-m_offset.hour());
+				update(flag::none, flag::zeroing);
+			}
+		} else {
+			f32 delta{};
+			if (m_input->any_held(key_t::Down, key_t::Left, key_t::S, key_t::A)) { delta += -1.0f; }
+			if (m_input->any_held(key_t::Up, key_t::Right, key_t::W, key_t::D)) { delta += 1.0f; }
+			add(translation_scale * delta * dt.count());
+		}
+	}
+
+  private:
+	using flags_t = u8;
+	enum flag : flags_t { none = 0, zeroing = 1 << 0 };
```

Comment:
This allows stuffing up to 8 flags into a single byte, whereas `bool`s would occupy a byte each. Another approach would be to force single bit storage via `u8 zeroing : 1`, but I prefer this.